vulnerabilities/scraper/lwn.py

```diff
@@ -0,0 +1,116 @@
+# Author: Islam Hiko (@EslamHiko)
+# Copyright (c) 2020 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+from bs4 import BeautifulSoup as bs
+from urllib.request import urlopen
+import re
+
+
+base_url = "https://lwn.net/"
+
+
+def extractPackageData(advisoryLink, dist, advisoryId):
+
+    content = urlopen(advisoryLink).read()
+    soup = bs(content, "html.parser")
+    text = soup.find('div', {'class': 'ArticleText'}).get_text()
+    phrases = text.split('\n')
+    cves = []
+    references = []
+    summary = ""
+    for i in range(len(phrases)):
+        words = phrases[i].split()
+        if phrases[i].startswith('Subject:'):
+            summary = phrases[i + 1].strip()
+        for word in words:
+            if word.startswith('CVE-') and word != 'CVE-ID':
+                cves.append(word)
+            elif word.startswith('https://') or word.startswith('http://'):
+                references.append(word)
+
+    cves = list(set(cves))
+
+    dist = re.sub(r'\W+', '', dist).replace('_', '').lower()
+
+    return {
+        'cve_ids': cves,
+        'references': references,
+        'summary': summary,
+        'advisory_id': advisoryId,
```

Comment:
I'm not sure how this is handled for Gentoo but by looking at the sample data given  
 ```
......
"summary": "[gentoo-announce] [ GLSA 201911-07 ] Mozilla Firefox: Multiple vulnerabilities",
    "advisory_id": "201911-07",
......
```
I think it will be more correct if  it looked something like this `"advisory_id":" GLSA 201911-07"` .  Ofcourse that would mean crafting separate logic for gentoo(mailing lists are painful)